WordPress/Classes/Utility/Media/MediaAssetExporter.swift

```diff
@@ -0,0 +1,236 @@
+            }
+        }
+
+        func toNSError() -> NSError {
+            return NSError(domain: _domain, code: _code, userInfo: [NSLocalizedDescriptionKey: String(describing: self)])
+        }
+    }
+
+    /// Helper method encapsulating exporting either an image or video.
+    ///
+    func exportData(forAsset asset: PHAsset, onCompletion: @escaping (AssetExport) -> Void, onError: @escaping (MediaExportError) -> Void) {
+        if asset.mediaType == .image {
+            exportImage(forAsset: asset, onCompletion: onCompletion, onError: onError)
+        } else if asset.mediaType == .video {
+            exportVideo(forAsset: asset, onCompletion: onCompletion, onError: onError)
+        } else {
+            onError(AssetExportError.unsupportedPHAssetMediaType)
+        }
+    }
+
+    fileprivate func exportImage(forAsset asset: PHAsset, onCompletion: @escaping (AssetExport) -> Void, onError: @escaping (MediaExportError) -> Void) {
+        do {
+            guard asset.mediaType == .image else {
+                throw AssetExportError.expectedPHAssetImageType
+            }
+
+            // Get the resource matching the type, to export.
+            let resources = PHAssetResource.assetResources(for: asset).filter({ $0.type == .photo })
+            guard let resource = resources.first else {
+                throw AssetExportError.unavailablePHAssetImageResource
+            }
+
+            if UTTypeEqual(resource.uniformTypeIdentifier as CFString, kUTTypeGIF) {
+                // Since this is a GIF, handle the export in it's own way.
+                exportGIF(forAsset: asset, resource: resource, onCompletion: onCompletion, onError: onError)
+                return
+            }
+
+            // Configure the options for requesting the image.
+            let options = PHImageRequestOptions()
+            options.version = .current
+            options.deliveryMode = .highQualityFormat
+            options.resizeMode = .exact
+            options.isNetworkAccessAllowed = true
+
+            // Configure the targetSize for PHImageManager to resize to.
+            let targetSize: CGSize
+            if let maximumImageSize = maximumImageSize {
+                targetSize = CGSize(width: maximumImageSize, height: maximumImageSize)
+            } else {
+                targetSize = PHImageManagerMaximumSize
+            }
+
+            // Configure an error handler for the image request.
+            let onImageRequestError: (Error?) -> Void = { (error) in
+                guard let error = error else {
+                    onError(AssetExportError.failedLoadingPHImageManagerRequest)
+                    return
+                }
+                onError(self.exporterErrorWith(error: error))
+            }
+
+            // Request the image.
+            let manager = PHImageManager.default()
```

Comment:
Just a thought could we make this a lazy variable? with the default value being PHImageManager.default(), but then it will allow user to change the ImageManager to another one if they need to cache requests.